Utils/Dataflow/095_datasetInfoAMI/amiDatasets.py

```diff
@@ -105,6 +114,9 @@ def process(stage, message):
     :type stage: pyDKB.dataflow.stage.ProcessorStage
     :param message: input message with data
     :type message: pyDKB.dataflow.communication.messages.JSONMessage
+
+    :return: False (message contains incorrect data) or True (otherwise)
```

Comment:
This line makes me think that `True` means "input data are correct", not "processing finished in a routine mode, without any (unexpected) problems", while `True`/`False` are meant to mean "processed successfully"/"failed to process message" (for whatever reason). Maybe for now it looks like the only reason to say "we failed, skip the message" is to have invalid input data, but I guess there is always a room for different unpleasant situations ;)
Can it be reworded somehow?

Reply:
Done.

> Maybe for now it looks like the only reason to say "we failed, skip the message" is to have invalid input data, but I guess there is always a room for different unpleasant situations ;)

Actually, we already have different situations - `amiPhysValues()` returns `True` or `False` depending on its results (however, at the moment it returns `False` when catching exception and when simply no data was found in AMI), so it may be logical to output this return instead of `True` at the end of `process()`. Will look into it.

Reply:
After some consideration I decided not to change anything further because the return value of `amiPhysValues()` is not used anywhere. When (and if) it will be used, we will decide on this point. Who knows, maybe we'll need to know whether the data was updated - in this case situations "something went wrong and nothing was done" and "no data was found in AMI" will indeed be the same for us.

Reply:
@Evildoor, 
Thank you.

---

I mark the conversation as "resolved" (since it *is* resolved), but to make things clear: my point was about `process()` function, not about *any* function with boolean return values. Any arbitrary taken function is allowed to have its own interpretation of boolean return values -- but `process()`, being an *implementation* of an abstract method, must follow the definition given in the library:
https://github.com/PanDAWMS/dkb/blob/66f68d10f80cefb39e197d3b405cba5233d15007/Utils/Dataflow/pyDKB/dataflow/stage/ProcessorStage.py#L278-L288

If it does not follow the definition, but is used as the method implementation, we will eventually face an unpredictable behaviour of a stage -- which we try to avoid using the common library.
Now it is only a matter of docstring, but following the inaccurate documentation the function might later be changed to something unhealthy.